spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/StructTypeImplicits.scala

```diff
@@ -413,4 +414,45 @@ object StructTypeImplicits {
     }
   }
 
+  implicit class StructTypeComparisonWhileIgnoringNullability(val schema: StructType)(implicit errorColumnName: String = ErrorMessageArray.defaultErrorColumnName) {
```

Comment:
I like the idea of such a function for the purposes. But would rather use `DataTypeEnhancements.isEquivalentDataType` enhanced to:
```scala
isEquivalentDataType(other: DataType, ignoreNullability: false, ignoreArrayContainsNull: false, ignoreMapValueContainsNull: false): Boolean
```